src/Command/Support.js

```diff
@@ -0,0 +1,318 @@
+                type,
+                description: this.config.support.types[type].description
+            }));
+            return message.channel.send(this.i18n.__mf(messages.help, {types}));
+        }
+
+        /* Handle the unique cases of "close" and "convert" */
+        tokens.shift();
+        const supportCategory = message.guild.channels.cache.get(this.config.support.category);
+        if (tokens.length > 0 && commandParameters.command === 'support'
+            && RoleDeterminer.isAdministrator(message.member)
+            && (tokens[0].toLowerCase() === 'close' || tokens[0].toLowerCase() === 'convert')) {
+            if (message.channel.parent.id !== supportCategory.id) {
+                return message.channel.send(this.i18n.__mf(messages.notASupportRoom));
+            }
+
+            if (tokens[0].toLowerCase() === 'close') {
+                return await this.processDeletion(message, commandParameters);
+            }
+            if (tokens[0].toLowerCase() === 'convert') {
+                return await this.processConversion(message, tokens);
+            }
+        }
+
+        /* Handle the case of no username provided */
+        if (!commandParameters.ign) {
+            if (commandParameters.command === 'support') {
+                return message.channel.send(this.i18n.__mf(messages.noIgnProvidedWithType, {
+                    command: commandParameters.command,
+                    type: commandParameters.supportType
+                }));
+            } else {
+                return message.channel.send(this.i18n.__mf(messages.noIgnProvided, {
+                    command: commandParameters.command
+                }));
+            }
+        }
+
+        /* Get the support type */
+        const typeKey = commandParameters.supportType;
+        const type = this.config.support.types[typeKey];
+        if (!type) {
+            const types = Object.keys(this.config.support.types).map(type => this.i18n.__mf(messages.type, {
+                type,
+                description: this.config.support.types[type].description
+            }));
+            const helpText = this.i18n.__mf(messages.help, {types});
+            return message.channel.send(this.i18n.__mf(messages.unknownType, {type: typeKey, help: helpText}));
+        }
+
+        /* Create the channel and add the correct people to it */
+        const supportChannel = await message.guild.channels.create(this.generateRoomName(typeKey), {
+            type: 'text',
+            parent: supportCategory,
+        })
+            .then(channel => channel.lockPermissions())
+            .then(channel => channel.updateOverwrite(message.author.id, {
+                ADD_REACTIONS: true,
+                ATTACH_FILES: true,
+                EMBED_LINKS: true,
+                READ_MESSAGE_HISTORY: true,
+                SEND_MESSAGES: true,
+                USE_EXTERNAL_EMOJIS: true,
+                SEND_TTS_MESSAGES: true,
```

Comment:
Support creator should not have TTS messages permission.  Nobody wants that, lol.